HW06.Rmd

```diff
@@ -19,14 +20,27 @@ Steps:
 
 From the RNA-Seq files, you only need the "Name" from one file and the "TPM" column from all the files. TPM stands for "transcripts per million" and is a common unit for normalized expression data. 
 ```{r load in data}
-#hint, using apply (specifically sapply) you can read in the data into a list and then bind the columns together. Or you can use a for loop too. 
 
-#you only need the 
+#Load R packages
+library(readr)
+library(dplyr)
+library(tidyverse)
+
+#Create a list of files within folder "RNA_Seq_processed"
+rna_file.list <- list.files("D:/BSDG_30901/Project/HW06/RNA_Seq_processed", recursive = TRUE, full.names = TRUE)
+
+#Use lapply to read files into a data frame list 
+rna_df.list <- sapply(rna_file.list, read_tsv, simplify = FALSE)
+
+#Combine columns together and rename name variable as gene_id
+GBM.transcripts <- bind_cols(rna_df.list)  %>%
+#keep "Name" from one file and the "TPM" column from all the files
+  select(gene_id = Name...1, starts_with("TPM")) 
 

```

Comment:
I am wondering that whether it will save some computational time if first select columns and then combine the data. (just my personal curiosity)I think the bind_cols() function is very useful and good to know for me!

Reply:
I think it should save some computation time if I first select columns and then combine the data sets. Thank you for the suggestion!